modules/twinklexfd.js

```diff
@@ -818,6 +818,100 @@ Twinkle.xfd.callbacks = {
 			Twinkle.xfd.callbacks.showPreview(form, venue, params);
 		}
 	},
+	/**
+	 * Unified handler for sending {{Xfd notice}} notifications
+	 * Also handles userspace logging
+	 * @param {object} params
+	 * @param {string} notifyTarget The user or page being notified
+	 * @param {boolean} [noLog=false] Whether to skip logging to userspace
+	 * XfD log, especially useful in cases in where multiple notifications
+	 * may be sent out (MfD, TfM, RfD)
+	 * @param {string} [actionName] Alternative description of the action
+	 * being undertaken. Required if not notifying a user talk page.
+	 */
+	notifyUser: function(params, notifyTarget, noLog, actionName) {
+		// Ensure items with User talk or no namespace prefix both end
+		// up at user talkspace as expected, but retain the
+		// prefix-less username for addToLog
+		notifyTarget = mw.Title.newFromText(notifyTarget, 3);
+		var targetNS = notifyTarget.getNamespaceId();
+		var usernameOrTarget = notifyTarget.getRelativeText(3);
+		notifyTarget = notifyTarget.toText();
+		if (targetNS === 3) {
+			// Disallow warning yourself
+			if (usernameOrTarget === mw.config.get('wgUserName')) {
+				Morebits.status.warn('You (' + usernameOrTarget + ') created this page; skipping user notification');
+
+				// if we thought we would notify someone but didn't,
+				// then jump to logging.
+				Twinkle.xfd.callbacks.addToLog(params, null);
+				return;
+			}
+			// Default is notifying the initial contributor, but MfD also
+			// notifies userspace page owner
+			actionName = actionName || 'Notifying initial contributor (' + usernameOrTarget + ')';
+		}
+
+		var notifytext = '\n{{subst:' + params.venue + ' notice';
+		// Venue-specific parameters
+		switch (params.venue) {
+			case 'afd':
+			case 'mfd':
+				notifytext += params.numbering !== '' ? '|order=&#32;' + params.numbering : '';
+				break;
+			case 'tfd':
+				if (params.xfdcat === 'tfm') {
+					notifytext = '\n{{subst:Tfm notice|2=' + params.tfdtarget;
+				}
+				if (mw.config.get('wgPageContentModel') === 'Scribunto') {
+					notifytext += '|module=yes';
+				}
+				break;
+			case 'cfd':
+				notifytext += '|action=' + params.action + (mw.config.get('wgNamespaceNumber') === 10 ? '|stub=yes' : '');
+				break;
+			default: // ffd, rfd
+				break;
+		}
+		// {{Tfd notice}} can't handle the namespace prefix
+		notifytext += '|1=' + (params.venue === 'tfd' ? mw.config.get('wgTitle') : Morebits.pageNameNorm) + '}} ~~~~';
```

Comment:
```suggestion
		notifytext += '|1=' + Morebits.pageNameNorm + '}} ~~~~';
```
I have fixed {{Tfd notice}} so that we don't need the special case.